src/main/java/com/synopsys/integration/alert/web/config/JobConfigController.java

```diff
@@ -224,6 +231,30 @@ public JobConfigController(JobConfigActions jobConfigActions, ResponseFactory re
         }
     }
 
+    @GetMapping("/validate")
+    public ResponseEntity<String> validateConfig() {
+        try {
+            List<PermissionKey> keys = new LinkedList<>();
+            for (Map.Entry<DescriptorKey, Descriptor> entry : descriptorMap.getDescriptorMap().entrySet()) {
```

Comment:
Since we're not really benefitting from the use of DescriptorMap here, you can change the passed variable to be a list of descriptors and then just loop through that. You can get the descriptorKey from the descriptor itself.